repos/system_upgrade/el7toel8/actors/mariadbcheck/libraries/library.py

```diff
@@ -0,0 +1,192 @@
+import os
+from leapp.models import InstalledRedHatSignedRPM
+from leapp import reporting
+from leapp.libraries.stdlib import CalledProcessError, api, run
+
+
+def parse_mycnf(config_file):
+    """ get used mysql options """
+    try:
+        output = run([
+            '/usr/bin/my_print_defaults',
+            '-c', config_file,
+            '--mysqld'
+        ], split=True)['stdout']
+    except CalledProcessError:
+        api.current_logger().warning('Unable to obtain MySQL local configuration.')
+        return None
+    return [line.split('=', 1) for line in output.splitlines()]
+
+
+def get_plugin_dir(config):
+    """ return plugin dir """
+
+    for entry in config:
```

Comment:
This would raise TypeErrir if config is None (which can happen in case MYSQL configuration can't be obtained, due to `return None` on L17).
Either return [] from parse_mycnf, or count with config=None possibility in get_plugin_dir.